internal/core/pipeline_config.go

```diff
@@ -0,0 +1,33 @@
+package core
+
+import (
+	"context"
+
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+// ConfigSync will evaluate the current hcl config for a given Pipeline and
+// upsert the proto version based on the current evaluation
+func (p *Pipeline) ConfigSync(ctx context.Context) error {
+	// TODO(briancain): In the future, we can sync config vars for pipelines here
+
+	// Sync the pipeline metadata
+	p.logger.Debug("evaluating pipeline configs for syncing")
+	pipelines, err := p.config.Config.PipelineProtos()
+	if err != nil {
+		return err
+	}
+
+	// TODO(briancain): do we need a Multi upsert?
```

Comment:
Seems like we might need a multi-pipeline Upsert? This could end up being a lot of API calls for a single config 🤔